ui/pages/confirmations/components/confirm/info/approve/hooks/use-is-nft.ts

```diff
@@ -0,0 +1,18 @@
+import { TransactionMeta } from '@metamask/transaction-controller';
+import { TokenStandard } from '../../../../../../../../shared/constants/transaction';
+import { useAsyncResult } from '../../../../../../../hooks/useAsyncResult';
+import { getTokenStandardAndDetails } from '../../../../../../../store/actions';
+
+export const useIsNFT = (
+  transactionMeta: TransactionMeta,
+): { isNFT: boolean; pending: boolean } => {
+  const { value, pending } = useAsyncResult(async () => {
+    return await getTokenStandardAndDetails(
+      transactionMeta?.txParams?.to as string,
+    );
+  }, [transactionMeta]);
+
+  const isNFT = value?.standard !== TokenStandard.ERC20;
```

Comment:
Minor, I appreciate we're only using this for labels currently, but one to keep an eye on as some wouldn't consider an ERC-1155 an NFT if it supported multiple of a token ID.